### PR TITLE
improvement(rangePicker): add max/min date support to range-picker

### PR DESCRIPTION
**Pull request by yashasvi13**
## Demo Link: [Storybook](https://604b209acbb07e0007155079--elastic-kare-88bfa0.netlify.app/?path=/story/components-inputs-date--documentation) 
To allow disabling dates after current date in Analytics
Added max and min selectable dates in date range picker

### CODE REVIEW

**Review by sidMulay — APPROVED** [review]

**Review by sidMulay — DISMISSED** [review]

File: src/components/Inputs/Date/__tests__/Date.basic.test.js [hunk 1]
```diff
@@ -45,6 +47,8 @@ const Component = ({
     numMonths={numMonths}
     withRange={withRange}
     withComparison={withComparison}
+    maxDate={maxDate}
```
Comment (swapnilmisal) [inline src/components/Inputs/Date/__tests__/Date.basic.test.js:50]: There could be four different variations to write the tests. for min and max data

File: src/components/Inputs/Date/components/RangeComparison/RangePicker.js [hunk 2]
```diff
@@ -213,6 +217,8 @@ RangePicker.defaultProps = {
   withComparison: false,
   numMonths: 2,
   prefixClassName: "",
+  maxDate: dayjs(),
```
Comment (swapnilmisal) [inline src/components/Inputs/Date/components/RangeComparison/RangePicker.js:220]: For a better developer experience,  if a developer do not pass min or max date, the selection of range should not be limited in any way.

File: src/components/Inputs/Date/Date.mdx [hunk 3]
```diff
@@ -35,6 +41,7 @@ import Date from "./Date";
       onChange={onChange}
       numMonths={2}
       withRange
+      maxDate={dayjs().add(7, "d")}
```
Comment (swapnilmisal) [inline src/components/Inputs/Date/Date.mdx:44]: Please add more variations for better developer experience.

**Review by swapnilmisal — CHANGES_REQUESTED** [review]

File: src/components/Inputs/Date/components/RangeComparison/RangePicker.js (hunk 2 above)
Comment (yashasvi13) [reply src/components/Inputs/Date/components/RangeComparison/RangePicker.js:220]: default value of max/min Date is already defined in the parent component as null, so it was not taking default value of its own component when max/min date is not passed, it simply takes null, so basically having default value of max/min in child component is useless

File: src/components/Inputs/Date/components/RangeComparison/RangePicker.js [hunk 4]
```diff
@@ -201,6 +203,8 @@ RangePicker.propTypes = {
   withComparison: PropTypes.bool,
   prefixClassName: PropTypes.string,
   numMonths: PropTypes.number,
+  maxDate: PropTypes.object,
+  minDate: PropTypes.object,
```
Comment (gopal-virtual) [inline src/components/Inputs/Date/components/RangeComparison/RangePicker.js:207]: call it `instanceOf(Date)`

**Review by gopal-virtual — CHANGES_REQUESTED** [review]

File: src/components/Inputs/Date/components/RangeComparison/RangePicker.js (hunk 4 above)
Comment (yashasvi13) [reply src/components/Inputs/Date/components/RangeComparison/RangePicker.js]: Done

**Review by gopal-virtual — DISMISSED** [review]

**Review by sidMulay — APPROVED** [review]

File: src/components/Inputs/Date/__tests__/Date.basic.test.js (hunk 1 above)
Comment (yashasvi13) [reply src/components/Inputs/Date/__tests__/Date.basic.test.js:50]: Done

**Review by swapnilmisal — APPROVED** [review]

**Review by gopal-virtual — APPROVED** [review]

**Comment by yashasvi13** [comment]
Reopening PR for checks

**Pull request merged**